### PR TITLE
[Authn] webportal logic to login

### PR DESCRIPTION
**Pull request by ydye**
#### The previous work:

webportal has the following logic to inject env.
```
{% if not cluster_cfg['authentication']['OIDC'] %}
        - name: AUTHN_METHOD
          value: basic
{% else %}
        - name: AUTHN_METHOD
          value: OIDC
{% endif %}
```
#### Next steps
Implement backend logic of login in restserver

### CODE REVIEW

File: src/webportal/src/app/home/index.jsx [hunk 1]
```diff
@@ -193,8 +193,14 @@ const Home = () => {
   );
 
   const showLoginModal = useCallback(
-    () => setLoginModal(true),
-    [],
+      () => {
+        if (config.authnMethod === 'basic') {
+          setLoginModal(true);
+        } else {
+          location.href=config.restServerUri + '/api/v1/auth/oidc/login?callback=' + encodeURIComponent(location.href);
```
Comment (Gerhut) [inline src/webportal/src/app/home/index.jsx]:
```suggestion
          location.href = config.restServerUri + '/api/v1/auth/oidc/login?callback=' + encodeURIComponent(location.href);
```

File: src/webportal/src/app/user/user-logout/user-logout.component.js [hunk 2]
```diff
@@ -15,14 +15,19 @@
...
   cookies.remove('user');
   cookies.remove('token');
   cookies.remove('admin');
   cookies.remove('hasGitHubPAT');
   cookies.remove('my-jobs');
-  window.location.replace('/index.html');
+  if (webportalConfig.authnMethod === 'basic') {
+    window.location.replace('/index.html');
+  } else {
+    location.href=webportalConfig.restServerUri+'/api/v1/auth/oidc/logout';
```
Comment (Gerhut) [inline src/webportal/src/app/user/user-logout/user-logout.component.js]:
```suggestion
    location.href = webportalConfig.restServerUri+'/api/v1/auth/oidc/logout';
```

File: src/webportal/src/app/home/index.jsx (hunk 1 above)
Comment (ydye) [reply src/webportal/src/app/home/index.jsx]: Done

File: src/webportal/src/app/user/user-logout/user-logout.component.js (hunk 2 above)
Comment (ydye) [reply src/webportal/src/app/user/user-logout/user-logout.component.js]: Done

**Pull request merged**